styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java

```diff
@@ -108,30 +118,29 @@ public void transitionInto(RunState state) {
     }
   }
 
-  private String dockerRunnerStart(RunState state) throws IOException {
-    final WorkflowInstance workflowInstance = state.workflowInstance();
+  private RunSpec createRunSpec(RunState state) throws ResourceNotFoundException {
     final Optional<ExecutionDescription> executionDescriptionOpt = state.data().executionDescription();
 
     final ExecutionDescription executionDescription = executionDescriptionOpt.orElseThrow(
         () -> new ResourceNotFoundException("Missing execution description for "
                                             + state.workflowInstance().toKey()));
 
+    final String executionId = state.data().executionId()
+        // Backwards compatibility fallback
+        .orElseGet(ExecutionDescriptionHandler::createExecutionId);
```

Comment:
Hard to think what will happen here... As we replay, we'll work with this real-looking unrelated ID?